### DISCUSSION TITLE
Проблема в работе виджета

### DISCUSSION THREAD

**Discussion by registreysn** [cdek-it/widget#25] [General]
Здравствуйте! В интернет-магазине используем данный виджет.
Вчера посетителям на сайте не удавалось выбрать офис доставки.
Сперва думали проблема с нашим сайтом.
Стали разбираться — обнаружилось, что сбой у самого виджета.
Работоспособность проверяли на сайте https://widget.cdek.ru/
С 3-х разных телефонов: iphone 6s, Samsung A7 (2018) , Poco X4 GT
На Poco X4 GT — офисы подгружаются, но с большой задержкой.
На  iphone 6s, Samsung A7 (2018) — всё зависает и не ужаётся выбрать нужный офис.
Видео с тестированием прилагаю к письму.
P.s. видео сжал, чтобы получилось прикрепить к сообщению.

https://github.com/cdek-it/widget/assets/33404745/bc17b52b-dd4e-4d1d-86bf-355ba0264fbc

https://github.com/cdek-it/widget/assets/33404745/fb4efd1f-077f-4ecb-b077-46b7d23c1252

**Comment by vermorag** [comment]
Добрый день. Это связано с тем, что для исполнения основных функций (подгрузки точек ПВЗ, расчета стоимости доставки и т.д.), Виджет обращается к хосту общих протоколов интеграции - api.cdek.ru. Начиная с понедельника, 25.12 примерно с 5 утра по МСК, периодически api.cdek.ru вместо 2-3 секунд может отвечать до 14-30 при запросе информации о точках ПВЗ. Нагрузка на модуль связана с высоким сезоном, в течение недели ожидаем стабилизацию времени ответа. Дополнительно обратили внимание на данный момент команде разработки общих протоколов и завели задачу на исследование.

**Reply by kipkaev** [reply]
Мы просто сохраняем себе эти данные на одни сутки, и не обращаемся в этот период к `api.cdek.ru`.
Проблемы есть с самой производительностью, особенно на мобильных устройствах, но это уж, я думаю, скорее к Яндексу с их картами.

**Reply by registreysn** [reply]
Можно по подробнее пожалуйста

**Reply by vermorag** [reply]
Вероятно, производительность может еще просаживаться из-за большого числа точек. С 4.0 версии доработаем полнофункциональное ограничение региона и должно стать по легче.

**Reply by registreysn** [reply]
Ограничение региона что подразумевает под собой? 
На текущий момент у меня на сайте и у других реализовано следующим образом:
- по IP определяется город, например Москва, при выборе пункта выдачи клиент видит все ближайшие офисы и например может выбрать офис в Апрелевке. 
Если речь про это ограничение ( клиент не будет видеть Апрелевку). 
То это не есть хорошо.

**Reply by vermorag** [reply]
Пользователь через параметр может задать (относительно  defaultLocation) какие пвз показывать - только в указанном городе, в регионе, в который входит указанный город, в стрене, в которую входит указанный город.

**Comment by registreysn** [comment]
Здравствуйте! Высокий сезон закончился, проблема с виджетом так и сохранилась.
Необходима ли какая-либо помощь с моей стороны для решения этой проблемы?

**Reply by vermorag** [reply]
Добрый день. Пока не могу дать конкретный ответ. После конца праздников, наша команда займется исследованием данного вопроса. Если у вас есть возможность - приложите профайлинг из браузера на подобии моего из https://github.com/cdek-it/widget/issues/18#issuecomment-1844076036, а также скриншот запросов из network (пример в https://github.com/cdek-it/widget/issues/16#issuecomment-1834429460), с указанием IP вашего сервера, на котором размещен service.php и с которого уходят запросы на api2.

**Comment by registreysn** [comment]
Здравствуйте! Речь ведь идëт про виджет, который находится на офф сайте сдек. 
https://widget.cdek.ru
Если он не работает как надо, то на остальных сайтах смысла тестировать нет. 
У меня, как и у других Ваших клиентов интернет-магазин. 
Из-за данной ошибки - клиенты не могут выбрать необходимый им офис доставки.
Соответственно они покупают в другом месте, на том же Вайлдберриз и т.д., где нет таких проблем. 
До 25.12.23 всë работало идеально. 
Стоит ли ждать решение данной проблемы или лучше не тратить время и переходить на другой сервис доставки?

**Reply by vermorag** [reply]
Добрый день. Коллегам из команды разработки передан запрос на доработку системы. Так как команды разработки имеют высокую степень загруженности, срок решения задачи может составлять от нескольких дней до нескольких недель. На данный момент, плановый срок выполнения - 24.01.24.

**Reply by vermorag** [reply]
Прошу обновиться до версии 3.9.0 и проверить.

**Comment by registreysn** [comment]
Здравствуйте! 
На Вашем сайте обновлено?
https://widget.cdek.ru/
Проблема только ухудшилась.
--
Отправлено из Mail.ru для Android среда, 24 января 2024г., 11:39 +03:00 от Alex Vi  ***@***.*** :

>Прошу обновиться до версии 3.9.0 и проверить.
>—
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>You are receiving this because you authored the thread. Message ID:  @ github . com>

К этому письму приложены ссылки на следующие файлы:

1. Screen_Recording_20240124-115040_Chrome.mp4 (49.1 Мб)

Ссылка для скачивания файлов: https://cloud.mail.ru/stock/nepBpiSVX8D9mUHu22VhP5eh

Файлы будут храниться до 22.07.2024

**Reply by vermorag** [reply]
Обновление на сайте будет в течении недели.

**Reply by registreysn** [reply]
Прошу обновить версию на сайте
https://widget.cdek.ru/ в более короткий срок, дабы исключить влияние каких-либо сторонних факторов, которые могут наблюдаться со стороны работы моего сайта. 

Текущая версия, которая сейчас стоит у меня на сайте : "кое-как" через раз, но работает. 
Рисковать и "Пробовать" обновить на версию, при установке которой могут всплыть непредвиденные обстоятельства - на решение которых потом может уйти несколько дней, к сожалению не могу себе позволить.

**Reply by registreysn** [reply]
Здравствуйте! Прошел месяц. Обновили на сайте?

**Reply by vermorag** [reply]
Добрый день. На сайте виджет обновляется автоматически в течении недели после выпуска новой версии.

**Comment by registreysn** [comment]
https://github.com/cdek-it/widget/assets/33404745/6d1bca70-455f-4d5b-ad40-647c685b846e

Ребят Вы серьëзно? 3 месяца прошло с момента, как сообщил Вам о баге. Ничего не изменилось.

**Reply by vermorag** [reply]
В настоящий момент, данная проблема прорабатывается и будет устранена с выходом новых версий карты.
